crates/bevy_ecs/src/reflect.rs

```diff
@@ -209,16 +210,14 @@ impl<C: Component + Reflect + FromWorld> FromType<C> for ReflectComponent {
                     .map(|c| c as &dyn Reflect)
             },
             reflect_mut: |world, entity| {
-                // SAFETY: reflect_mut is an unsafe function pointer used by `reflect_unchecked_mut` which promises to never
-                // produce aliasing mutable references, and reflect_mut, which has mutable world access
+                // SAFETY: reflect_mut is an unsafe function pointer used by
+                // 1. `reflect_unchecked_mut` which must be called with an InteriorMutableWorld with access the the component `C` on the `entity`, and
+                // 2. reflect_mut, which has mutable world access
```

Comment:
Nit: Typo and punctuation consistency
```suggestion
                // 1. `reflect_unchecked_mut` which must be called with an `InteriorMutableWorld` with access to the component `C` on the `entity`, and
                // 2. `reflect_mut`, which has mutable world access
```